content/Developer_Program/WebAuthn_Starter_Kit/Adaptive_Multi-Factor_Authentication.adoc

```diff
@@ -0,0 +1,169 @@
+
+=== Overview
+
+When implementing an Adaptive Multi-Factor Authentication with WebAuth, there are impacts on everything from new account creation to login to credential management. Many of these changes will reduce user friction, and allow for strong authentication to be offered alongside increased ease of use. For clarity, each element will be discussed in the sections below.
+
+==== User Registration
+[%header,cols="1,1,1,1,1"]
+|===
+| *WebAuthn authenticator* | *Examples* | *Provide identifier* | *Register a WebAuthn credential* | *Create U2F Password*
+
+| FIDO2
+a| 
+* Yubikey 5 Series
+* YubiKey BIO
+* Face ID
+a|image::Images/group-4.png[]
+a|image::Images/group-4.png[]
+|
+
+| U2F
+a| 
+* Android Biometrics and Security Key
+* YubiKey 4 Series
+a|image::Images/group-4.png[]
+a|image::Images/group-4.png[]
+a|image::Images/group-4.png[]
+
+| Usernameless - FIDO2 with a Resident Credential
+a| 
+* YubiKey 5 Series
+a|image::Images/group-4.png[]
+a|image::Images/group-4.png[]
+|
+|===
+
+As the user accesses the WebAuthn Starter Kit for the first time they are prompted for their username, which represents a unique identifier for the user’s account. The username can be defined by the user, or based on a user’s email or phone number. For the WebAuthn Starter Kit, the username is separate from the user’s email or phone number to demonstrate a basic flow and provide code focused on the WebAuthn implementation. Note that using an identifier such as username or telephone number may reduce privacy, as users can enter known values to see if an account is associated with them - it is up to the architect to decide if this is a concern for their service.
+ 
+Optionally, the username can be stored on a local client or via a cookie or token for a web app to remove this requirement after the user has logged in once. If this method is implemented, options should be provided in the login page to allow a user to enter a new username.
+ 
+Once a username is submitted to the app, a decision point is reached. If the username already exists in the user store, the user is directed to the user login flow. If the username does not exist, the user is directed to the new account creation flow.
+ 
+If a username does not already exist, the user is directed to the new account creation flow. The first step is to start the create account process - displaying the username and offering the user an opportunity to return to the previous dialog if they entered the name incorrectly.
```

Comment:
These statements are out of date. We refactored the UI to provide an explicit Sign up and Sign in page that the user must navigate to.